### PR TITLE
fix: use junit data to determine openshift-suite pass/fail status

### PR DESCRIPTION
**Pull request by whereswaldon**
This commit does several things:
- Fixes all of the tests to be configured as cluster admin
- Stops overriding the user account in use away from cluster admin
- Changes the way that we determine failure from exit status to
  parsing the junit XML. Unparsable xml, lack of xml, or failures
  in XML will cause the test to fail.

Signed-off-by: Chris Waldon <cwaldon@redhat.com>

### CODE REVIEW

File: pkg/e2e/openshift/images.go [hunk 1]
```diff
@@ -42,6 +78,11 @@ var _ = ginkgo.Describe(imageRegistryTestName, func() {
 		results, err := r.RetrieveResults()
```
Comment (jeefy) [inline pkg/e2e/openshift/images.go:78]: What are your thoughts on including both calling `ensurePassingXML` as well as checking that it's valid and passing as part of `r.RetrieveResults()` ?

I feel like that makes tests less redundant but I could see value in being explicit in each case.

Comment (whereswaldon) [reply pkg/e2e/openshift/images.go:78]: I think that's a good idea *if and only if* there is no usage of the helper that doesn't produce JUnit result data. If we always expect this, then we should absolutely do that.

**Comment by jeefy** [comment]
/lgtm
What a pain

**Pull request merged**